qa/rpc-tests/test_framework/util.py

```diff
@@ -78,8 +90,16 @@ def initialize_chain(test_dir):
     bitcoind and bitcoin-cli must be in search path.
     """
 
-    if not os.path.isdir(os.path.join("cache", "node0")):
-        devnull = open("/dev/null", "w+")
+    if ( not os.path.isdir(os.path.join("cache", "node0"))
+             or not os.path.isdir(os.path.join("cache", "node1")) 
+             or not os.path.isdir(os.path.join("cache", "node2")) 
+             or not os.path.isdir(os.path.join("cache", "node3")) ):
+
+        #find and delete old cache directories if any exist
+        for i in range(4):
+	    if os.path.isdir(os.path.join("cache", "node"+str(i))): shutil.rmtree(os.path.join("cache", "node"+str(i)))
```

Comment:
Made the code updates you requested and pushed to GitHub...

On 16/08/2015 12:44 PM, MarcoFalke wrote:
>
> In qa/rpc-tests/test_framework/util.py 
> <https://github.com/bitcoin/bitcoin/pull/6548#discussion_r37149332>:
>
> > @@ -78,8 +90,16 @@ def initialize_chain(test_dir):
> >      bitcoind and bitcoin-cli must be in search path.
> >      """
> >
> > -    if not os.path.isdir(os.path.join("cache", "node0")):
> > -        devnull = open("/dev/null", "w+")
> > +    if ( not os.path.isdir(os.path.join("cache", "node0"))
> > +             or not os.path.isdir(os.path.join("cache", "node1"))
> > +             or not os.path.isdir(os.path.join("cache", "node2"))
> > +             or not os.path.isdir(os.path.join("cache", "node3")) ):
> > +
> > +        #find and delete old cache directories if any exist
> > +        for i in range(4):
> > +	    if os.path.isdir(os.path.join("cache", "node"+str(i))): shutil.rmtree(os.path.join("cache", "node"+str(i)))
>
> So you only want to delete existing cache dirs when at least one cache 
> dir needs to be created?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/bitcoin/bitcoin/pull/6548/files#r37149332>.
>